src/sss_client/nss_mc_common.c

```diff
@@ -178,6 +179,38 @@ static void sss_nss_mc_destroy_ctx(struct sss_cli_mc_ctx *ctx)
     /* `mutex` and `active_threads` should be left intact */
 }
 
+#define MC_FD_OFFSET 100
+
+static int make_high_fd(int fd)
+{
+    if (fd < 0) {
+        return fd;
+    }
+
+#ifdef F_DUPFD
+    struct rlimit rlim;
+    int target_fd;
+    int new_fd;
+
+    if (getrlimit(RLIMIT_NOFILE, &rlim) != 0) {
+        return fd;
+    }
+
+    target_fd = (int)(rlim.rlim_cur - MC_FD_OFFSET);
+    if (target_fd <= STDERR_FILENO) {
+         return fd;
+    }
+
+#define _POSIX_C_SOURCE 200809L
+    new_fd = fcntl(fd, F_DUPFD_CLOEXEC, target_fd);
+    if (new_fd != -1) {
+        return new_fd;
```

Comment:
i think you need to `close(fd);` here to prevent an fd leak.

Reply:
sure, thanks